s3.tf

```diff
@@ -89,26 +93,43 @@ resource "aws_s3_bucket_inventory" "apiary_bucket" {
 }
 
 resource "aws_s3_bucket_public_access_block" "apiary_bucket" {
-  count  = var.s3_block_public_access == true ? "${length(local.apiary_data_buckets)}" : 0
-  bucket = "${aws_s3_bucket.apiary_data_bucket.*.id[count.index]}"
+  for_each = var.s3_block_public_access == true ? {
+    for schema in local.schemas_info : "${schema["schema_name"]}" => schema
+  } : {}
+  bucket = aws_s3_bucket.apiary_data_bucket[each.key].id
 
   block_public_acls   = true
   block_public_policy = true
   ignore_public_acls  = true
 }
 
 resource "aws_s3_bucket_notification" "data_events" {
-  count  = "${var.enable_data_events == "" ? 0 : length(local.apiary_data_buckets)}"
-  bucket = "${aws_s3_bucket.apiary_data_bucket.*.id[count.index]}"
+  for_each = var.enable_data_events == "" ? {} : {
+    for schema in local.schemas_info : "${schema["schema_name"]}" => schema if lookup(schema, "enable_data_event_queue", "0") == "0"
+  }
+  bucket = aws_s3_bucket.apiary_data_bucket[each.key].id
 
   topic {
-    topic_arn = "${aws_sns_topic.apiary_data_events.*.arn[count.index]}"
+    topic_arn = aws_sns_topic.apiary_data_events[each.key].arn
     events    = ["s3:ObjectCreated:*", "s3:ObjectRemoved:*"]
   }
 }
 
+resource "aws_s3_bucket_notification" "data_queue_events" {
+  for_each = { for schema in local.schemas_info : "${schema["schema_name"]}" => schema if lookup(schema, "enable_data_event_queue", "0") == "1" }
+  bucket   = aws_s3_bucket.apiary_data_bucket[each.key].id
+
+  queue {
+    queue_arn     = aws_sqs_queue.apiary_data_event_queue[0].arn
+    events        = ["s3:ObjectCreated:*", "s3:ObjectRemoved:*"]
+  }
+}
+
+
 resource "aws_s3_bucket_metric" "paid_metrics" {
-  count  = "${var.enable_s3_paid_metrics == "" ? 0 : length(local.apiary_data_buckets)}"
-  bucket = "${aws_s3_bucket.apiary_data_bucket.*.id[count.index]}"
+  for_each = var.enable_s3_paid_metrics == "" ? {} : {
```

Comment:
since we are creating a breaking change, can you we update these variables to use actual `bool` values `true/false`

Reply:
Currently, all the breaking changes I have created so far can be mitigated by running the migrate script, and the user doesn't have to do anything to their app code except update TF 0.12 version.

Changing all the 0/1 variables to boolean would create real breaking changes that the migration script can't handle for the user.  Is that OK?  If so, I can make those changes.

Reply:
generally, it is better to introduce breaking changes when updating major version. may we be can merge this change but not release, then create another PR with all breaking changes for variables.